migrate plugin to Typescript and generate modular packages

stklcode opened the pull request:
This is quite a major update.
The Plugin and the project structure are updated to get closer to the _Chartist_ v1 ecosystem.

I was wondering if the changes should be part of a new, shared repo, but I couldn't manage to start a discussion upstream. My goal is not to create and maintain yet another fork, so maybe we want to move this project in the proposed direction...

Changes:
* Migrate sources from _JavaScript_ to _Typescript_
* Use _Chartist_ v1 API instead of the hacky workaround in v1
* Remove "donutSolid" Pie Chart support
   They are no longer available in _Chartist_.
* Remove "currency" and "currencyFormat" options
   Kind of tricky to use and they can easily be replaced with a one-line callback.

Build Process:
* Replace _Grunt_ with _Yarn_ tasks
* Build ESM, CJS and UMD versions using _Rollup.js_
* Extended unit tests
* (Re-)Introduce _ESLint_ and _Prettier_

mahlm commented:
Tooltips are such a standard feature, I wonder if the PR could be forked to the chartist org: https://github.com/chartist-js.

stklcode commented:
I tried to start a discussion about moving common plugins to the @chartist-js org: https://github.com/chartist-js/chartist/discussions/1391

LukBukkit commented:
Sorry, for the long radio silence. The last months were very busy for me. I'll review your contributions in the following days or weeks. Did your code works well in production or do you've noticed any possible drawbacks?

stklcode commented:
Just rebased the PR on top of the latest changes, so it can be applied cleanly again.

I'm using the updated lib in one project with various line charts and it served my need since I introduced the changes. (it's in an admin UI, so I currently don't have a public site to link here). I have not tested every single feature.

Don't know if there is any more complete example page that could be simply migrated to see if everything works as expected. We could probably build one.

LukBukkit commented:
I'm sorry that I haven't found the time so far to review the pull request. My time available for open-source projects has decreased significantly in the last year. Are you using the plugin in production, and if so, would you be interested in maintaining it in the future?
